### PR TITLE
Simplify RequestLineFrom

### PR DESCRIPTION
**Pull request by Sammers21**
Methods should  not throw any checked exceptions

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/http/issues/34#issuecomment-584674518))

@g4s8 This is what I know about this job in [CT2E6TK9B](https://www.0crat.com/p/CT2E6TK9B), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#34](https://github.com/artipie/http/issues/34) is [in scope](http://www.zerocracy.com/policy.html#14) for 43min
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fhttp%2334%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@0crat assign @paulodamaso

**Comment by 0crat** [comment]
> @0crat assign @paulodamaso  ([here](https://github.com/artipie/http/issues/34#issuecomment-584700305))

@g4s8 This pull request [#34](https://github.com/artipie/http/issues/34) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/d8385a8d-64f0-43d7-9702-99d0fca3a472); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java [hunk 1]
```diff
@@ -50,61 +46,49 @@
...
-    private final Text line;
-
-    /**
-     * New reqiest line from string.
-     * @param line Request line string
-     */
-    public RequestLineFrom(final String line) {
-        this(new TextOf(line));
-    }
+    private final String line;
```
Comment (paulodamaso) [inline src/main/java/com/artipie/http/rq/RequestLineFrom.java]: @Sammers21 Keep the `Text` field

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java [hunk 2]
```diff
@@ -50,61 +46,49 @@
...
      * @param idx Part index
      * @return Part string
-     * @throws IOException In case of IO error
      * @todo #12:30min Check idx value to avoid ArrayIndexOfBoundsException.
      *  Valid HTTP request line must contains 3 parts which can be splitted by
      *  whitespace char. If line doesn't contain 3 parts, let's throw an exception.
      */
-    private String part(final int idx) throws IOException {
-        return new IoChecked<>(this.line::asString).value().trim().split("\\s")[idx];
+    private String part(final int idx) {
```
Comment (paulodamaso) [inline src/main/java/com/artipie/http/rq/RequestLineFrom.java]: @Sammers21 Use `cactoos` decorators to trim and split the `Text`. And, if you want, you can get rid of this method and create constants for each piece of the information (METHOD = 0, URI = 1 and VERSION = 2) and you won't need to use this method

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java [hunk 3]
```diff
@@ -50,61 +46,49 @@
...
-        this(new TextOf(line));
-    }
+    private final String line;
 
     /**
      * Primary ctor.
      * @param line HTTP request line
      */
-    public RequestLineFrom(final Text line) {
+    public RequestLineFrom(final String line) {
```
Comment (paulodamaso) [inline src/main/java/com/artipie/http/rq/RequestLineFrom.java]: @olegmoz Keep using `Text`

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java (hunk 1 above)
Comment (Sammers21) [reply src/main/java/com/artipie/http/rq/RequestLineFrom.java]: No. That makes things more complex for no reason.

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java (hunk 2 above)
Comment (Sammers21) [reply src/main/java/com/artipie/http/rq/RequestLineFrom.java]: I'm not using `Text`

File: src/main/java/com/artipie/http/rq/RequestLineFrom.java (hunk 3 above)
Comment (Sammers21) [reply src/main/java/com/artipie/http/rq/RequestLineFrom.java]: No, the idea is to use `String`

**Comment by Sammers21** [comment]
@paulodamaso, `RequestLineFrom` whose primary task is to split a string into 3 parts and obtain them. This is a very simple task and to use `Text`, `IoChecked` for that is overkill. Moreover,  throwing `IOException` is unreasonable.

**Comment by g4s8** [comment]
@paulodamaso I approve these changes: we needed `Text` here when we passed `Request` object here, but the design was change, so we don't have request object anymore, so the `Text` wrapping here is not needed.

**Comment by paulodamaso** [comment]
@g4s8 What about lazy evaluation and making this code less procedural?

**Comment by paulodamaso** [comment]
@Sammers21 Please at least put these magic numbers into constants

**Comment by g4s8** [comment]
@paulodamaso we a `String` as request line in `Slice` interface, it can't be lazy by design

**Comment by paulodamaso** [comment]
@g4s8 Okay. Well, I just have a comment about turning the magic number into constants, if you think that it is not necessary, then you can merge

**Comment by Sammers21** [comment]
> @g4s8 Okay. Well, I just have a comment about turning the magic number into constants, if you think that it is not necessary, then you can merge

@paulodamaso , they are not a part of the diff

**Comment by paulodamaso** [comment]
@Sammers21 well, it is enough then. Good to merge

**Comment by paulodamaso** [comment]
@g4s8 Please merge, then

**Comment by 0crat** [comment]
Job was finished in 46 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by g4s8** [comment]
@rultor release, tag=`0.2.1`

**Comment by atefaz** [comment]
@rultor hello

**Pull request merged**